code/datums/manufacturing_requirements.dm

```diff
@@ -1,216 +1,334 @@
+			if (MATCH_EXACT)
+				return (M.getMaterialFlags(M) == src.material_flags)
+
+/datum/manufacturing_requirement/match_flags/metal
+	name = "Metallic"
+	id = "metal_flag"
+	material_flags = MATERIAL_METAL
+
+/datum/manufacturing_requirement/match_flags/wood
+	name = "Wood"
+	id = "wood_flag"
+	material_flags = MATERIAL_WOOD
 
-/datum/manufacturing_requirement/match_property/rubber
+/datum/manufacturing_requirement/match_flags/rubber
 	name = "Rubber"
 	id = "rubber"
 	material_flags = MATERIAL_RUBBER
 
-/datum/manufacturing_requirement/match_property/wood
-	name = "Wood"
-	id = "wood"
-	material_flags = MATERIAL_WOOD
+/datum/manufacturing_requirement/match_flags/organic_or_rubber
+	name = "Organic or Rubber"
+	id = "organic_or_rubber"
+	material_flags = MATERIAL_ORGANIC | MATERIAL_RUBBER
+
+/datum/manufacturing_requirement/match_flags/fabric
+	name = "Fabric"
+	id = "fabric"
+	material_flags = MATERIAL_RUBBER | MATERIAL_ORGANIC | MATERIAL_CLOTH
+
+/datum/manufacturing_requirement/match_flags/crystal
+	name = "Crystal"
+	id = "crystal"
+	material_flags = MATERIAL_CRYSTAL
+
+/datum/manufacturing_requirement/match_flags/energy
+	name = "Energy Source"
+	id = "energy"
+	material_flags = MATERIAL_ENERGY
+
+/datum/manufacturing_requirement/match_flags/insulated
+	name = "Insulative Material"
+	id = "insulative_flags"
+	material_flags = MATERIAL_CLOTH | MATERIAL_RUBBER
+
+#undef MATCH_ANY
+#undef MATCH_ALL
+#undef MATCH_EXACT
+
+ABSTRACT_TYPE(/datum/manufacturing_requirement/match_subtypes)
+/datum/manufacturing_requirement/match_subtypes
+	VAR_PROTECTED/match_typepath //! The parent type to use for istype() checks
+
+	#ifdef CHECK_MORE_RUNTIMES
+	New()
+		. = ..()
+		if (isnull(src.match_typepath) || !ispath(src.match_typepath))
+			CRASH("[src] has invalid match_typepath [src.match_typepath], it must be non-null and a valid path (not an instance!)")
+	#endif
+
+	is_match(var/datum/material/M)
+		. = ..()
+		if (!.) return
+		return istype(M, src.match_typepath)
+
+/datum/manufacturing_requirement/match_subtypes/gemstone
+	name = "Gemstone"
+	id = "gemstone"
+	match_typepath = /datum/material/crystal/gemstone
+
+/// Manufacturing requirements which check several conditions at once.
+ABSTRACT_TYPE(/datum/manufacturing_requirement/mixed)
+/datum/manufacturing_requirement/mixed
+	VAR_PROTECTED/list/requirement_ids = list() //! A list of requirement IDs to populate requirements with their instances in the cache
+	VAR_PROTECTED/list/datum/manufacturing_requirement/requirements = list() //! A list of requirements which must all be satisfied for this to return TRUE
+
+	/// Resolve the requirement paths to instances in the cache.
+	New()
+		. = ..()
+		var/list/datum/manufacturing_requirement/requirement_instances = list()
+
+		for (var/requirement_id in src.requirement_ids)
+			requirement_instances += getManufacturingRequirement(requirement_id)
+		src.requirements = requirement_instances
```

Comment:
Why create a new list then assign it and not just operate on the target list? Note my change assumes that `src.requirements` is not used by `..()`.

Also edited to avoid one of the `list()` on the type definition (rather than within `New`).
```suggestion
	VAR_PROTECTED/list/datum/manufacturing_requirement/requirements = null //! A list of requirements which must all be satisfied for this to return TRUE

	/// Resolve the requirement paths to instances in the cache.
	New()
		. = ..()
		src.requirements = new
		for (var/requirement_id as anything in src.requirement_ids)
			src.requirements += getManufacturingRequirement(requirement_id)
```